src/cmd/device-protection.js

```diff
@@ -325,25 +303,26 @@ module.exports = class DeviceProtectionCommands extends CLICommandBase {
 	 * @param {Function} fn - The function to execute with the device.
 	 * @returns {Promise<*>} The result of the function execution.
 	 */
-	async _withDevice({ putDeviceBackInDfuMode, spinner }, fn) {
-		try {
-			await this._getUsbDevice(this.device);
-			return await this.ui.showBusySpinnerUntilResolved(spinner, (async () => {
-				const deviceWasInDfuMode = this.device.isInDfuMode;
-				if (deviceWasInDfuMode) {
+	async _withDevice({ putDeviceBackInDfuMode, spinner, supportSafeMode }, fn) {
```

Comment:
Make sure to add new argments to the docs. I can't tell what `supportSafeMode` is mean to do.